test/message.spec.js

```diff
@@ -0,0 +1,126 @@
+var Promise = require("bluebird");
+var expect = require("chai").expect;
+var supertest = require("supertest-promised");
+var parseXML = require("xml-parser");
+var find = require("lodash/find");
+var app = require("../app");
+var SurveyResponse = require("../models/SurveyResponse");
+var agent = supertest(app);
+
+describe("POST /message", function() {
+  beforeEach(function() {
+    return SurveyResponse.remove({})
+    .then(() => console.log("### delete all"))
+  });
+
+  it("Should return a greeting/question on first survey attempt.", function() {
+    return agent.post("/message")
+      .type("form")
+      .send({
+        From: "+15555555555",
+        Body: "Message body",
+      })
+      .expect("Content-Type", /text\/xml/)
+      .expect(200)
+      .expect(function(res) {
+        var doc = parseXML(res.text);
+
+        var messageEl = find(doc.root.children, function(el) {
```

Comment:
Conceptually both the element `Message` and the message are part of the same assertion. I'm expecting the node message to contain the text `"Thank you for..."`.

For situations like this one, I would prefer to use _xpath_ which provides the syntax for querying a node and get the content.

The obvious benefit will be that you'll write a single expectation that will check both, structure and content. What do you think?